### ISSUE TITLE
[BUG] Intermittent inability to edit links

### ISSUE THREAD

**Issue by amandamartin-dev** [EddieHubCommunity/LinkFree#7851]
### Description

This bug does not happen on all links and may not be worth investigating yet unless it is found to be widespread, but during an edit session on the link in the video below, I am faced with the following issues:

I am unable to add additional text to the field
I can delete a few characters - am still unable to type in the field
I must clear the entire field to be able to edit as expected.

Working on Chrome: Version 114.0.5735.133 (Official Build) (x86_64)
and MacOS Ventura 13.2.1 

https://github.com/EddieHubCommunity/LinkFree/assets/97615019/527823fe-7339-44b5-88a5-00bc4560328c

**Comment by dan-mba** [comment]
The input for the link URL has the setting `maxLength="256"`.
Could your link be longer than 256 characters?

**Comment by amandamartin-dev** [comment]
> The input for the link URL has the setting `maxLength="256"`. Could your link be longer than 256 characters?

This is my fault for using the wrong language, it's not the URL that I was editing but the "name". Regardless, my name entry in that case is 84 characters (incl spaces)

**Comment by dan-mba** [comment]
> > The input for the link URL has the setting `maxLength="256"`. Could your link be longer than 256 characters?
> 
> This is my fault for using the wrong language, it's not the URL that I was editing but the "name". Regardless, my name entry in that case is 84 characters (incl spaces)

The Display Name field has the setting `maxLength="64"`

**Comment by amandamartin-dev** [comment]
> The Display Name field has the setting `maxLength="64"`

Sounds like that's the issue then.  When you say setting, I'm assuming you mean at the level of the database so does @eddiejaoude need to adjust this or are you able to?

I'm assuming we should also add an issue for feedback in the UI if the editor is trying to do something that is restricted or is at a character limit.

**Comment by dan-mba** [comment]
> > The Display Name field has the setting `maxLength="64"`
> 
> 
> 
> Sounds like that's the issue then.  When you say setting, I'm assuming you mean at the level of the database

No, `maxLength` is set on the html input element

It is in the code [here](https://github.com/EddieHubCommunity/LinkFree/blob/d2393f7989ea861b1c5ecee1f89eed5f8dc3478f/pages/account/manage/link/%5B%5B...data%5D%5D.js#L211)

**Comment by amandamartin-dev** [comment]
> No, `maxLength` is set on the html input element
> 
> It is in the code [here](https://github.com/EddieHubCommunity/LinkFree/blob/d2393f7989ea861b1c5ecee1f89eed5f8dc3478f/pages/account/manage/link/%5B%5B...data%5D%5D.js#L211)

Ah, perfect. I assume opening it up to 256 would be fine but since that was a choice I'll wait to set this to ready to dev.

**Comment by eddiejaoude** [comment]
Great discussion 👍 

We can increase it if needed, in addition to Dan's code link, we would need to increase the server side validation in the model here https://github.com/EddieHubCommunity/LinkFree/blob/main/models/Link.js#L20

Is 256 too big for a link name?

**Comment by eddiejaoude** [comment]
Yes the DB has limits on the models, but they were introduced recently if I remember correctly. We don't have any other issues for validation. Can we list what needs to be done and maybe include it in this issue or we can raise another one, but yes having a list will help

**Comment by amandamartin-dev** [comment]
@eddiejaoude I can open it as a discussion.  There are a few ways to approach and we will want to keep accessibility in mind as well.

**Comment by SaraJaoude** [comment]
@jss530 I will assign this to you. 

As to the scope: please increase the name input field for `link` to 128 and also the `name` field in the link model.

**Comment by jss530** [comment]
PR can be found [here](https://github.com/EddieHubCommunity/BioDrop/pull/8856).
